python/lsst/pipe/tasks/matchBackgrounds.py

```diff
@@ -19,683 +19,627 @@
-            Results as a struct with attributes:
-
-            ``backgroundInfoList``
-                A `list` of `pipeBase.Struct`, one per exposure in expRefList,
-                each of which contains these fields:
-                - ``isReference``: This is the reference exposure (only one
-                                   returned Struct will contain True for this
-                                   value, unless the ref exposure is listed multiple times).
-                - ``backgroundModel``: Differential background model
-                                       (afw.Math.Background or afw.Math.Approximate).
-                                       Add this to the science exposure to match the reference exposure.
-                - ``fitRMS``: The RMS of the fit. This is the sqrt(mean(residuals**2)).
-                - ``matchedMSE``: The MSE of the reference and matched images:
-                                  mean((refImage - matchedSciImage)**2);
-                    should be comparable to difference image's mean variance.
-                - ``diffImVar``: The mean variance of the difference image.
-                All fields except isReference will be None if isReference True or the fit failed.
+        result : `~lsst.afw.math.BackgroundList`, `~lsst.afw.image.Exposure`
+            Differential background models and associated background-matched
+            images.
 
         Raises
         ------
         RuntimeError
             Raised if an exposure does not exist on disk.
         """
-        numExp = len(expRefList)
-        if numExp < 1:
-            raise pipeBase.TaskError("No exposures to match")
-
-        if expDatasetType is None:
-            raise pipeBase.TaskError("Must specify expDatasetType")
-
-        if imageScalerList is None:
-            self.log.info("imageScalerList is None; no scaling will be performed")
-            imageScalerList = [None] * numExp
-
-        if len(expRefList) != len(imageScalerList):
-            raise RuntimeError("len(expRefList) = %s != %s = len(imageScalerList)" %
-                               (len(expRefList), len(imageScalerList)))
-
-        refInd = None
-        if refExpDataRef is None:
-            # select the best reference exposure from expRefList
-            refInd = self.selectRefExposure(
-                expRefList=expRefList,
-                imageScalerList=imageScalerList,
-                expDatasetType=expDatasetType,
-            )
-            refExpDataRef = expRefList[refInd]
-            refImageScaler = imageScalerList[refInd]
+        # TODO: matching currently done in fluence, not surface brightness,
+        # units.  Current solution for filling in non-overlapping regions
+        # between ref and other images is fitting a plane, but this fails as
+        # plane is typically buried beneath warping Jacobian pattern.  Unsure
+        # how to resolve this issue, but it's critical to resolve for BG-
+        # matching to be viable.
```

Comment:
I think the answer involves warping the inverse of the `backgroundToPhotometricRatio` image (see `CalibrateImageTask` connections), or when that is not available for an instrument, a detector-coordinate image of ones (which wouldn't be ones after it's warped).

But to use it you'd need to find a way to use a difference of two splines scaled by the two inverse ratios as your basis functions, and I don't see a way to do that without reimplementing the spline fitter.